src/app/static/src/app/components/forms/CustomForm.vue

```diff
@@ -0,0 +1,46 @@
+<template>
+    <b-form :ref="id" class="custom-form">
+        <div v-for="section in form.controlSections">
+            <h3>{{section.label}}</h3>
+            <p class="text-muted">{{section.description}}</p>
+            <control-group v-for="group in section.controlGroups" :group="group"></control-group>
+        </div>
+        <button class="btn btn-red" v-on:click="submit">Submit</button>
+    </b-form>
+</template>
+
+<script lang="ts">
+
+    import Vue from "vue";
+    import {BForm} from "bootstrap-vue";
+    import {formMeta} from "./fakeFormMeta";
+    import ControlGroup from "./ControlGroup.vue";
+    import {Dictionary} from "vuex";
+
+    export default Vue.extend({
+        name: "CustomForm",
```

Comment:
One slightly annoying thing about using real html tags and not jsx is that we can't name our custom components things like "form" because <form> will always be interpreted as the html tag. Any suggestions for a good name for this component?

Reply:
Would MetaForm be too meta? DataForm? DynamicForm? Don't know, can't really think of anything that isn't either potentially misleading or too long!

Reply:
Maybe DynamicForm is the most natural.